modules/getting-started/pages/quickstart-dotnet27-aspnetcore31-visualstudio-first-query-cb65.adoc

```diff
@@ -0,0 +1,154 @@
+----
+
+== Step 2: ASP.NET Core Configuration
+
+There are two steps to configure your new ASP.NET Core application.
+
+First, add a "Couchbase" section to the appsettings.json file. If you've followed along with the xref:quickstart-docker-image-manual.adoc[10-minute Couchbase Docker Container Configuration], then edit your appsettings.json file to add a Couchbase section as follows:
+
+[source,JavaScript]
+----
+include::{sourcedir}/quickstart-aspnetcore31-cb65/appsettings.json[]
+----
+
+The next step is to setup Couchbase in Startup.cs.
+
+To do this, first add `services.AddCouchbase(Configuration.GetSection("Couchbase"));` to the `ConfigureServices` method. Make sure to close the connection as well. You can do this in `Configure` with `IHostApplicationLifetime`. Here is the complete Startup.cs:
+
+[source,JavaScript]
+----
+include::{sourcedir}/quickstart-aspnetcore31-cb65/Startup.cs[]
+----
+
+At this point, your ASP.NET Core application is now configured to use Couchbase.
+
+== Step 3: Create a new document
+
+Let's create two controller actions. These will correspond to two API endpoints: one will create a document and one will query a document.
+
+You can start with the `WeatherForecastController` that Visual Studio automatically generated, or you can create your own. In either case, create a constructor that has an `IBucketProvider` parameter. Use the bucketProvider to get an instance of a Couchbase bucket: `_bucket = bucketProvider.GetBucket("default")`.
+
+[source,C#,indent=0]
+----
+include::{sourcedir}/quickstart-aspnetcore31-cb65/Controllers/WeatherForecastController.cs[tag=ctor]
+----
+
+Use this bucket object to insert plain C# objects into Couchbase. They will be serialized into JSON and stored with a key: `_bucket.Insert(key, forecast);`. Create an `InsertData` POST method like so:
+
+[source,C#,indent=0]
+----
+include::{sourcedir}/quickstart-aspnetcore31-cb65/Controllers/WeatherForecastController.cs[tag=insert]
+----
+
+Compile and run your ASP.NET Core program locally with CTRL+F5. Use cURL or Postman or the HTTP tool of your choice to create a POST with JSON content to the specified route. The body of the post should look something like this:
+
+[source,JavaScript]
+----
+{
+	"Date" : "2020-04-14",
+	"TemperatureC" : 25,
+	"Summary" : "Today's weather is partly cloudy and warm."
+}
+----
+
+When the request is successful, the response will contain the generated ID:
+
+[source]
+----
+Inserted forecast with ID: d28a19d4-0c37-447f-b0c0-66a7f2af3c97
+----
+
+Now a single document should appear in your Couchbase bucket. Make a note of this key; you'll need it for the next step. 
+
+image::bucketOneDocument.png[Template select in Visual Studio]
+
+== Step 4: Query the document
+
+Create a second controller action called `GetData`. This will be for an API endpoint to retrieve a document given a key.
+
+For this tutorial, we'll write a N1QL (SQL for JSON) query to retrieve the document:
+
+[source,SQL]
+----
```

Comment:
We want this to be a secondary lookup, not a primary key lookup.  SELECT * FROM default WHERE <attribute> ...

Reply:
This occurred to me after talking with @deniswsrosa this morning. Not sure what I was thinking 😜, but I've changed it to use temperatureC instead of the META().id